### PR TITLE
fix: projects/ti_to_ti README with evaluation results and examples

### PR DESCRIPTION
**Pull request by htlou**
# Description

Describe your changes in detail.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #1314520` if this solves the issue #15213

- [x] I have raised an issue to propose this change ([required](https://github.com/PKU-Alignment/align-anything/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/PKU-Alignment/align-anything/blob/HEAD/.github/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: projects/ti_to_ti/pre_tokenize_parallel_example.py [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+    for p in processes:
+        p.join()
+    
+    output_paths = list(output_paths)
+    
+    all_data = []
+    for path in output_paths:
+        data = torch.load(path)
+        all_data.append(data)  # 假设每个文件中的数据本身就是一个列表
```
Comment (cby-pku) [inline projects/ti_to_ti/pre_tokenize_parallel_example.py:309]: remove the note

Comment (htlou) [reply projects/ti_to_ti/pre_tokenize_parallel_example.py]: fixed in the https://github.com/PKU-Alignment/align-anything/pull/41/commits/63f0f76733e3ca55563bffdf1fcb0512364a70ff commit

File: projects/ti_to_ti/pre_tokenize_parallel_example.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    split_token: str = 'ASSISTANT:'
+    separator: str = '###'
+    # print(raw_sample)
+    input_text = raw_sample['question']
+    output_text = raw_sample['response']
+    input_img = raw_sample['image_url']
+    output_img = raw_sample['output_image_url']
+    
+    if isinstance(input_img, str):
+    # 单个图像
```
Comment (Gaiejj) [inline projects/ti_to_ti/pre_tokenize_parallel_example.py:109]: remove this comment

File: projects/ti_to_ti/README.md [hunk 3]
```diff
@@ -107,4 +113,47 @@ You can set the `INPUT_TYPE` in the `script.sh` to `interleaved-compare` and run
...
+
+We trained the lm_head of Chameleon-7B model using the full data from [laisonart](huggingface.co/datasets/fantasyfish/laion-art), and acquired a AA-Chameleon-7B model (AA refers to Align-Anything) with image generation capabilities. Then, we used our text-image interleaved input & output dataset (20k) to finetune this model, acquiring a AA-Chameleon-7B+, which is much better in text-image interleaved i/o task. The model is trained for 3 epochs with a batch size of 2 and a learning rate of 5e-05 (similar to meta's setting in [their paper](https://ai.meta.com/blog/generative-ai-text-images-cm3leon)). 
+
+### GPT-4o evaluation
+
+We compared the performance of AA-Chameleon-7B+ with Anole-7b model on a text-image interleaved i/o dataset (with a length of 500). Here's the results:
+
+| win | lose | equal |
+|-------|-------|-------|
+| 240 | 175 | 85 |
```
Comment (cby-pku) [inline projects/ti_to_ti/README.md:128]: Show the Win Rate, not the number of results

File: projects/ti_to_ti/pre_tokenize_parallel_example.py (hunk 2 above)
Comment (htlou) [reply projects/ti_to_ti/pre_tokenize_parallel_example.py]: fixed in https://github.com/PKU-Alignment/align-anything/pull/41/commits/25f0b8d01ede5f8e0e5cb058a3607d356ae46724 commit

File: projects/ti_to_ti/README.md (hunk 3 above)
Comment (htlou) [reply projects/ti_to_ti/README.md]: fixed in https://github.com/PKU-Alignment/align-anything/pull/41/commits/883bf36ca6c370fca200220a8a1166d320823c7d commit

File: projects/ti_to_ti/README.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+python pre_tokenize_example.py
+```
+
+If you are dealing with a large dataset, you can use `pre_tokenize_parallel_example.py` to speed up the process in multiple threads:
+
+```bash
+python pre_tokenize_parallel_example.py
+```
+
+### Chameleon Finetuning
```
Comment (XuyaoWang) [inline projects/ti_to_ti/README.md:35]: Change to `Model Finetuning`

File: projects/ti_to_ti/README.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+> Output text: <image>I have applied a vintage filter to the bathroom image to give it a retro look. The process involved adjusting the color temperature to add a slight cool tone, reducing the saturation to give it a more muted appearance, and increasing the contrast to enhance the vintage feel.
+
+> Output image: ![image](assets/3e93f869-0a61-4be3-adda-8f7fb9d1e99f.png)
+
+Example #2
+
+Input text: Create a detailed recipe for making the pastry shown in the image, including ingredients, preparation steps, and baking instructions. <image>
+
+Input image: ![image](assets/000000191297.jpg)
```
Comment (zmsn-2077) [inline projects/ti_to_ti/README.md:145]: 这个放在assets/ti_to_ti/xxx下面呢

File: projects/ti_to_ti/pre_tokenize_parallel_example.py [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+        
+        labels = return_dict['input_ids'].clone()
+        # mask non-assistant input
+        labels[: len(prompt_dict['input_ids'][0]) - 1] = IGNORE_INDEX
+        return_dict['labels'] = labels
+
+        return_dict['pixel_values'] = text_dict['pixel_values']
+        
+        # print(f"Input id shape: {return_dict['input_ids'].shape}")
+        # print(f"Pixel value shape: {return_dict['pixel_values'].shape}")
```
Comment (zmsn-2077) [inline projects/ti_to_ti/pre_tokenize_parallel_example.py:221]: delete

File: projects/ti_to_ti/pre_tokenize_parallel_example.py [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+    num_img = 0
+    num_text = 0
+    for piece in tqdm(input_data, desc=f"Processing on GPU {gpu}"):
+        formatted_sample = format_sample(piece)
+        preprocessed_sample, label_len, flag = preprocess(tokenizer, processor, formatted_sample)
+        if flag:
+            num_img += 1
+        else:
+            num_text += 1
+        # torch.set_printoptions(threshold=torch.inf)
```
Comment (zmsn-2077) [inline projects/ti_to_ti/pre_tokenize_parallel_example.py:242]: delete

Comment (htlou) [reply projects/ti_to_ti/pre_tokenize_parallel_example.py]: fixed in https://github.com/PKU-Alignment/align-anything/pull/41/commits/068a8bbbaf4a27c191a98f19007a7f354d922d7b

File: projects/ti_to_ti/pre_tokenize_parallel_example.py (hunk 6 above)
Comment (htlou) [reply projects/ti_to_ti/pre_tokenize_parallel_example.py]: fixed in https://github.com/PKU-Alignment/align-anything/pull/41/commits/068a8bbbaf4a27c191a98f19007a7f354d922d7b

File: projects/ti_to_ti/README.md (hunk 4 above)
Comment (htlou) [reply projects/ti_to_ti/README.md]: fixed in https://github.com/PKU-Alignment/align-anything/pull/41/commits/068a8bbbaf4a27c191a98f19007a7f354d922d7b

File: projects/ti_to_ti/README.md (hunk 5 above)
Comment (htlou) [reply projects/ti_to_ti/README.md]: fixed in https://github.com/PKU-Alignment/align-anything/pull/41/commits/139bdc2e3b908aa00105b7c78b462475af71740b

**Comment by htlou** [comment]
All the previous comments are fixed in newly added commits, request another review.

**Review by XuyaoWang — APPROVED** [review]
LGTM.

**Review by zmsn-2077 — APPROVED** [review]
LGTM
